File: backend/src/main/java/de/snickit/rezeptebackend/entities/pk/RecipeStepKey.java
package de.snickit.rezeptebackend.entities.pk;

import de.snickit.rezeptebackend.entities.Recipe;

import javax.persistence.Column;
import java.io.Serializable;

public class RecipeStepKey implements Serializable {

    private Recipe recipeId;

    private Long stepId;
}

File: backend/src/main/kotlin/de/snickit/recipe/backend/persistence/pk/RecipeStepKey.kt
package de.snickit.recipe.backend.persistence.pk

import de.snickit.recipe.backend.persistence.entities.Recipe
import java.io.Serializable
import javax.persistence.*

@Embeddable
data class RecipeStepKey (

    @ManyToOne
    @JoinColumn(name = "recipeId", foreignKey = ForeignKey(name = "recipe_step_recipe_fk"))
    val recipe: Recipe? = null,

    val stepId: Int = recipe?.recipeSteps?.size?.plus(1) ?: 1

) : Serializable
